src/System.Net.Security/tests/FunctionalTests/SslStreamSniTest.cs

```diff
@@ -70,29 +156,29 @@ public void SslStream_NoSniFromClient_CallbackReturnsNull()
                 };
 
                 var cts = new CancellationTokenSource();
-                Assert.Throws<AuthenticationException>(WithAggregateExceptionUnwrapping(() =>
-                    server.AuthenticateAsServerAsync(options, cts.Token).Wait()
+                await Assert.ThrowsAsync<AuthenticationException>(WithAggregateExceptionUnwrapping(async () =>
+                    await server.AuthenticateAsServerAsync(options, cts.Token)
                 ));
 
                 // to break connection so that client is not waiting
                 server.Dispose();
 
                 Assert.Equal(1, timesCallbackCalled);
 
-                clientJob.Wait();
+                await clientJob;
             },
             (object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors) =>
             {
                 return true;
             });
         }
 
-        private static Action WithAggregateExceptionUnwrapping(Action a)
+        private static Func<Task> WithAggregateExceptionUnwrapping(Func<Task> a)
         {
-            return () => {
+            return async () => {
                 try
                 {
-                    a();
+                    await a();
                 }
                 catch (AggregateException e)
```

Comment:
This catch block is likely dead code now.  You can leave it to avoid churning more than is necessary at this point, but whereas Task.Wait() wraps the one or more exceptions in the task into an AggregateException, await just propagates the first exception.

Reply:
Good to know, thanks